src/Sylius/Component/Product/Model/Product.php

```diff
@@ -26,9 +26,14 @@
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  * @author Gonzalo Vilaseca <gvilaseca@reiss.co.uk>
  */
-class Product extends AbstractTranslatable implements ProductInterface
+class Product implements ProductInterface
 {
-    use SoftDeletableTrait, TimestampableTrait, ToggleableTrait;
+    use SoftDeletableTrait;
+    use TimestampableTrait;
+    use ToggleableTrait;
+    use TranslatableTrait {
+        __construct as translatableConstruct;
```

Comment:
As it is the constructor's name discussion resulted in `initializeTranslationsCollection`.
IMO initializing it in the class would be a bit weird as the field itself, and all methods using it are virtually in the trait.